src/Lecture1.hs

```diff
@@ -94,7 +98,14 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+
+subString :: Int -> Int -> [a] -> [a]
+subString start end str
+    | start     > end = []
+    | end       < 0   = []
+    | otherwise = drop start' (take (end'+1) str)
```

Comment:
This is a neat implementation! 😎 

I'm already happy with it 🤗 However, it's possible to implement it more efficiently by dropping first and taking the required number of elements after that. I think it will be more efficient. But this is because I know how `drop` and `take` are implemented internally 😅 These both functions traverse the list until they reach the point of stopping.

So, for example, if you're given a string of 100 elements and you want to have a substring from index 70 to 99, your existing implementation will do 170 operations (99 + 1 for taking the entire list and then 70 for dropping the prefix) while dropping 70 first and taking the remaining 30 will require only 100 operations total.

But of course, I haven't explained the internals of `take` and `drop` in the first lecture. And you're not expected to know such internal implementation details at this point. Therefore, your implementation is pretty elegant 🆒

Reply:
I should have thought of it cause haskell has the same list implementation like erlang(my main language). I have rewrote the solution